.github/workflows/build-package.yaml

```
@@ -0,0 +1,238 @@
+# This automation builds 3p packages based on a PR
+
+name: Build 3P Packages
+
+on:
+  pull_request:
+    branches:
+      - main
+      - development
+    paths:
+      - 'package_build_list_host_*.json'
+
+jobs:
+  detect-changes:
+    name: Detecting changes in PR to build
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.detect-platform.outputs.matrix }}
+    
+    steps:
+    - name: Checkout 3P source repo
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 10
+          
+    - name: Get package and platform from JSON changes
+      id: detect-platform
+      run: |
+        CHANGED_FILES=$(git diff ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }} --name-only)
+        # Construct the package and os into a json string to be consumed by Github Actions runners
+        JSON="{\"include\":["
+        for FILE in $CHANGED_FILES; do
+        	if [[ $FILE == package_build_list_host_* ]]; then
+        		PLATFORM=$(echo $FILE | sed -n 's/package_build_list_host_\(.*\).json/\1/p')
+        		case $PLATFORM in
+        		linux*)
+        			OS_RUNNER="ubuntu-latest"
+        			;;
+        		windows)
+        			OS_RUNNER="windows-2019"
+        			;;
+        		darwin)
+        			OS_RUNNER="macos-latest"
+        			;;
+        		*)
+        			OS_RUNNER="windows-latest" # default
+        			;;
+        		esac
+        
+        		DIFF=$(git diff ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }} --no-ext-diff --unified=0 \
+                    --exit-code -a --no-prefix -- $FILE | egrep "^\+" | grep Scripts)
+
+            if [[ $? -ne 0 ]]; then
+              echo No valid build change found. Exiting with non-zero
+            fi
+            
+            PACKAGE=$(echo $DIFF | cut -d'"' -f2)
+        
+        		JSONline="{\"package\": \"$PACKAGE\", \"os\": \"$OS_RUNNER\"},"
+        		if [[ "$JSON" != *"$JSONline"* ]]; then
+        			JSON="$JSON$JSONline"
+        		fi
+        	fi
+        done
+
+        # Remove last "," and add closing brackets
+        if [[ $JSON == *, ]]; then
+          JSON="${JSON%?}"
+        fi
+        JSON="$JSON]}"
+        echo $JSON
+
+        # Set output
+        echo "matrix=$( echo "$JSON" )" >> $GITHUB_OUTPUT

```

There are more tabs here as well
```suggestion
            if [[ $? -ne 0 ]]; then
              echo No valid build change found. Exiting with non-zero
            fi
            
            PACKAGE=$(echo $DIFF | cut -d'"' -f2)
        
            JSONline="{\"package\": \"$PACKAGE\", \"os\": \"$OS_RUNNER\"},"
            if [[ "$JSON" != *"$JSONline"* ]]; then
              JSON="$JSON$JSONline"
            fi
          fi
        done

        # Remove last "," and add closing brackets
        if [[ $JSON == *, ]]; then
          JSON="${JSON%?}"
        fi
        JSON="$JSON]}"
        echo $JSON

        # Set output
        echo "matrix=$( echo "$JSON" )" >> $GITHUB_OUTPUT

```